File: Documentation/5.4/Samples/python/ClientApi/Operations/Maintenance/Indexes/GetIndex.py
from typing import List

from ravendb import GetIndexOperation, GetIndexesOperation, IndexDefinition
from ravendb.documents.operations.definitions import MaintenanceOperation

from examples_base import ExampleBase


class GetIndex(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_get_index(self):
        with self.embedded_server.get_document_store("GetIndex") as store:
            # region get_index
            # Define the get index operation, pass the index name
            get_index_op = GetIndexOperation("Orders/Totals")

            # Execute the operation by passing it to maintenance.send
            index = store.maintenance.send(get_index_op)

            # Access the index definition
            state = index.state
            lock_mode = index.lock_mode
            deployment_mode = index.deployment_mode
            # etc.

            # endregion

            # region get_indexes
            # Define the get indexes operation
            # Pass number of indexes to skip & number of indexes to retrieve
            get_index_op = GetIndexesOperation(0, 10)

            # Execute the operation by passing it to maintenance.send
            indexes = store.maintenance.send(get_index_op)

            # indexes will contain the first 10 indexes, alphabetically ordered by index name
            # Access an index definition from the resulting list:
            name = indexes[0].name
            state = indexes[0].state
            lock_mode = indexes[0].lock_mode
            deployment_mode = indexes[0].deployment_mode
            # etc.
            # endregion

    class Foo:
        # region get_index_syntax

        class GetIndexOperation(MaintenanceOperation[IndexDefinition]):
            def __init__(self, index_name: str): ...

        # endregion
        # region get_indexes_syntax
        class GetIndexesOperation(MaintenanceOperation[List[IndexDefinition]]):
            def __init__(self, start: int, page_size: int): ...

        # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Maintenance/Indexes/GetIndex.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Indexes\IndexDefinition;
use RavenDB\Documents\Indexes\IndexDefinitionArray;
use RavenDB\Documents\Operations\Indexes\GetIndexesOperation;
use RavenDB\Documents\Operations\Indexes\GetIndexOperation;

class GetIndex
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region get_index

            // Define the get index operation, pass the index name
            $getIndexOp = new GetIndexOperation("Orders/Totals");

            // Execute the operation by passing it to Maintenance.Send
            /** @var IndexDefinition $index */
            $index = $store->maintenance()->send($getIndexOp);

            // Access the index definition
            $state = $index->getState();
            $lockMode = $index->getLockMode();
            $deploymentMode = $index->getDeploymentMode();
            // etc.

            # endregion
        } finally {
            $store->close();
        }

        $store = new DocumentStore();
        try {
            # region get_indexes
            // Define the get indexes operation
            // Pass number of indexes to skip & number of indexes to retrieve
            $getIndexesOp = new GetIndexesOperation(0, 10);

            // Execute the operation by passing it to Maintenance.Send
            /** @var IndexDefinitionArray $indexes */
            $indexes = $store->maintenance()->send($getIndexesOp);

            // indexes will contain the first 10 indexes, alphabetically ordered by index name
            // Access an index definition from the resulting list:
            $name = $indexes[0]->getName();
            $state = $indexes[0]->getState();
            $lockMode = $indexes[0]->getLockMode();
            $deploymentMode = $indexes[0]->getDeploymentMode();
            // etc.
            # endregion
        } finally {
            $store->close();
        }
    }
}

/*
interface IFoo
{
    # region get_index_syntax
    GetIndexOperation(?string $indexName)
    # endregion

    # region get_indexes_syntax
    GetIndexesOperation(int $start, int $pageSize)
    # endregion
}
*/
